docs/maintainers/adr/0002-use-dart-sass-java-for-compilation.md

```diff
@@ -0,0 +1,47 @@
+# Use dart-sass-java for Sass compilation using Dart Sass
+
+* Status: accepted
+* Date: 2024-01-06
+
+Technical Story: [PLATUI-3423](https://jira.tools.tax.service.gov.uk/browse/PLATUI-3423)
+
+## Context and Problem Statement
+
+There are numerous different tools and libraries that will perform the compilation of Sass to CSS using Dart Sass. These
+are documented in the [Dart Sass docs](https://sass-lang.com/dart-sass/). These includes:
+* Dart Sass’s stand-alone command-line executable
+* Javascript library via `npm`
+* Java library `sass-embedded-host` (from [dart-sass-java](https://github.com/larsgrefer/dart-sass-java/))
+
+PlatUI wanted to investigate the above approaches, to see which would be the best to use in our Sass compilation plugin.
+
+## Decision Drivers
+
+* The need to minimise additional work by service developers, in terms of both writing additional code and installing new
+  tools or frameworks
+* The need for any solution to work as part of the automated build pipelines
+* The risk for any specific third party solution of code "rot", i.e. support being dropped outside of MDTP (as seen with
+  [sbt-sassify](https://github.com/irundaia/sbt-sassify))
+
+## Considered Options
+
+* Option 1: Dart Sass’s stand-alone command-line executable
+* Option 2: Javascript library via `npm`
+* Option 3: Java library `sass-embedded-host` (from [dart-sass-java](https://github.com/larsgrefer/dart-sass-java/))
+
+## Decision Outcome
+
+Chosen option 3: "Java library `sass-embedded-host`"
+
+## Pros and Cons of the Options
+* Pro of `sass-embedded-host`: Java does require installation on a local development machine, but that will already be a
+  prerequisite for Scala developers on MDTP. Using either `npm` or the standalone CLI executable would require additional downloads to any individual developer 
+  machine to run locally. Whilst the `sbt-sass-compiler` plugin could initiate this download and installation,
+  it still presents a potential risk to PlatUI in terms of increased support queries, version management of frameworks
+  such as Node or `npm`.
+* Pro of `sass-embedded-host`: Using the standalone CLI executable would require additional download and installation to build Jenkins, which is a risk
+  that would need to be discussed and signed off by Build & Deploy, which is not the case with `dart-sass-java`.
```

Comment:
with this, I think we wouldn't ask to add it to the build container, but include the compiled binary as a resource in the plugin and then unpack it to a temporary directory when we wanted to use it

like how sass-embedded-host includes it https://github.com/larsgrefer/dart-sass-java/blob/766248e026234ea23293fa2e7bb052dd7b16ca26/sass-embedded-bundled/build.gradle#L17C1-L42C1

it defines a custom ivy source that's a the dart-sass github url - not sure if that's an ivy thing we could adapt in future if we wanted or a gradle thing

Reply:
@oscarduignan Ah yes, I was thinking it would within the temporary directory of a specific build (not suggesting it would be added to the container globally), but I thought we'd still want sign-off from B&D, plus we may need to get the download URL allow listed to be accessible from Jenkins.

Reply:
@oscarduignan Should I be more explicit with that bullet point?

Reply:
I think it's all good, I was forgetting that it would need to download it during build of plugin, I was thinking "it won't download it, it will be packaged in the library" but that's that it wouldn't be downloaded when people use the library but the build of the library would need to download it - though because it's github it might already be allowlisted (though also we might be blocked from pulling from github still, and if so would need to be mirrored to artefactory)